### ISSUE TITLE
Adding a configuration-driven interface

### ISSUE THREAD

**Issue by jscanvic** [deepinv/deepinv#496]
High-level deep learning libraries like `ultralytics` feature configuration-driven interfaces for training and fine-tuning neural networks on custom data. Instead of tweaking a fully-fledged training script, a third-party program is distributed with the library and it can be used along with configuration files to train pre-existing models on custom data. At the moment, the library only supports programmatic usage which can deter less technical users from using the library. I suggest we add support for this use case.

A practical use case might look like:

```bash
python -m deepinv train --config ./config.yaml
```

```yaml
# config.yaml
dataset: ./dataset.h5
loss: EILoss
epochs: 100
model: UNet
out_dir: ./weights
...
```

**How**

The implementation can be as simple as a basic CLI wrapper over the existing trainer.

**Comment by Andrewwango** [comment]
I think this is a great idea, and I think it would be pretty trivial to implement, and it could be a good opportunity to make the QuickStart even clearer.

On this note, are there any other "recipes" you envisage? Because many users don't train at all - the trainer represents only one subclass of image reconstruction methods?

**Comment by jscanvic** [comment]
We might want to have a fairly general inference script as well. Something along the lines of: here's a bunch of measurements in a directory / h5 file or whatever, use this specific model as per the configuration (pre-trained network, optimization scheme), compute the estimates and put them into some directory / h5 file. Also possibly write useful things for debugging and validating, like the value of the loss function at each step for TV-denoising.
